### DISCUSSION TITLE
How to push to a fork's GHCR in a PR check?

### DISCUSSION THREAD

**Discussion by Twixes** [community/community#27348] [Actions]
Hey! At PostHog we're using GHCR in PR checks as an optimization of our E2E tests (which require a built Docker image of our app).
This works with no issues on PRs from our org members, but there is a big issue in PRs from external contributors: The `GITHUB_TOKEN` there only allows read-only access to _our_ repo, and not read-write access to the fork repo. This means we can't push to GHCR at all. Is there a way around this?
Specific actions we're using: `docker/metadata-action@v3`,  `docker/login-action@v1`, `docker/build-push-action@v2`.

**Comment by hazzadous** [comment]
For what it's worth we're gone with sharing via GitHub Action artifacts instead but still using ghcr.io for layer caching purposes, details can be found on https://github.com/PostHog/posthog/pull/11262/files
